packages/contracts/src/core/dao/DAO.sol

```diff
@@ -122,6 +154,21 @@ contract DAO is
         __PermissionManager_init(_initialOwner);
     }
 
+    /// @notice Initializes the DAO after an upgrade from a previous version.
+    /// @param previousProtocolVersion The protocol version number of the previous DAO implementation contract this upgrade is transitioning from.
+    function initializeUpgradeFrom(
+        uint8[3] calldata previousProtocolVersion
+    ) external reinitializer(2) {
+        if (previousProtocolVersion[0] != 1 || previousProtocolVersion[1] >= 3) {
+            revert ProtocolVersionUpgradeNotSupported(previousProtocolVersion);
+        }
+
+        // All versions before v1.3.0 must initialize the newly added `_reentrancyStatus`.
+        if (previousProtocolVersion[1] < 3) {
```

Comment:
If we leave the "if" block above, this check would always be true

Reply:
>  I would also revert in case someone upgrades from 1.3.0 to 1.3.0 because this is also fishy / wrong.

The proposal I made does nothing in this case, to save gas, assuming that you can't reinitialize the same build (OZ)

Reply:
> Lastly, is it on purpose that you want to allow people to upgrade from 0.Y.Z versions by writing previousProtocolVersion[0] > 1?

Not sure I follow. This is a check to prevent upgrades coming from higher releases

Reply:
```solidity
if (previousProtocolVersion[0] != 1) revert();
else if (previousProtocolVersion[1] >= 3) revert();

// build upgrades
if (previousProtocolVersion[1] < 3) {
  _reentrancyStatus = _NOT_ENTERED
}
```

Reply:
I think  having`||` and returning the protocol version in the error is better, no? I removed the redundant `if` here: 51c831cae41a847295fb1f75ebd4fa781575a82a